extensions/opc/src/opc.cpp

```diff
@@ -220,10 +220,31 @@ NodeData Client::getNodeData(const UA_ReferenceDescription *ref, const std::stri
       nodedata.attributes["NodeID type"] = "numeric";
     }
     nodedata.attributes["Browsename"] = browsename;
+    // TODO: Base path is always ""
     nodedata.attributes["Full path"] = basePath + "/" + browsename;
     nodedata.dataTypeID = UA_TYPES_COUNT;
     UA_Variant* var = UA_Variant_new();
     if(UA_Client_readValueAttribute(client_, ref->nodeId.nodeId, var) == UA_STATUSCODE_GOOD && var->type != NULL && var->data != NULL) {
+      // Because the timestamps are eliminated in readValueAttribute for simplification
```

Comment:
If possible, we should only do this if we are in lazy mode, otherwise this is unnecessary.

Reply:
Non-layz does the same thing in the background, the 3rd party provides a simplified API function to hide this.
I don't mind having a bit more complex but single code path in case it has no performance drawbacks.

Reply:
Yes, this code part is a trimmed down version of `UA_readValueAttribute` with a slight modification. Despite checking lazy mode before running might be better, I then have to pass the lazy flag from fetchopc processor to this function, which I am not quite into.